src/main/java/io/jenkins/plugins/gitlabbranchsource/credentials/PersonalAccessTokenImpl.java

```diff
@@ -0,0 +1,92 @@
+package io.jenkins.plugins.gitlabbranchsource.credentials;
+
+import com.cloudbees.plugins.credentials.CredentialsDescriptor;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+
+/**
+ * Default implementation of {@link PersonalAccessToken} for use by {@link Jenkins} {@link CredentialsProvider}
+ * instances that store {@link Secret} locally.
+ */
+
+public class PersonalAccessTokenImpl extends BaseStandardCredentials implements PersonalAccessToken {
+
+    /**
+     * Our token.
+     */
+    @NonNull
+    private final Secret token;
+
+    /**
+     * Constructor.
+     *
+     * @param scope       the credentials scope.
+     * @param id          the credentials id.
+     * @param description the description of the token.
+     * @param token       the token itself (will be passed through {@link Secret#fromString(String)})
+     */
+    @DataBoundConstructor
+    public PersonalAccessTokenImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
```

Comment:
Another super nit, but this might be easier to read if there was one parameter per line